erpnext/controllers/selling_controller.py

```diff
@@ -548,3 +548,50 @@ def update_project_billing_and_sales(self):
 			doc.set_gross_margin(update=True)
 			doc.set_status(update=True)
 			doc.notify_update()
+
+	@frappe.whitelist()
+	def set_rate_as_cost(self):
+		if not has_valuation_read_permission():
+			frappe.throw('User not allowed to read valuation rate')
+
+		for item in self.items:
+			if item.item_code:
+				cost_rate = self.get_rate_as_cost(item)
+				item.rate = cost_rate
+				item.discount_percentage = 0
+				item.margin_rate_or_amount = 0
+
+		self.calculate_taxes_and_totals()
+
+	def get_rate_as_cost(self, item):
+		qty = -1 * flt(item.qty)
+		if item.delivery_note and item.delivery_note_item:
+			sum_of_stock_value_difference = frappe.db.sql("""
+				SELECT SUM(stock_value_difference)
+				FROM `tabStock Ledger Entry` 
+				WHERE voucher_type = 'Delivery Note' AND voucher_no = %s AND voucher_detail_no = %s
+			""", (item.delivery_note, item.delivery_note_item))
+
+			sum_of_stock_value_difference = flt(sum_of_stock_value_difference[0][0])
```

Comment:
Accessing [0] on a None problem still not resolved

Do `sum_of_stock_value_difference = flt(sum_of_stock_value_difference[0][0]) if sum_of_stock_value_difference else 0`